### DISCUSSION TITLE
How to utilize "group:allNonMajor" and "group:nodeJs" presets simultaneously?

### DISCUSSION THREAD

**Discussion by chalkygames123** [renovatebot/renovate#11572] [Request Help] [answered]
I'd like to see the PRs separated into two groups, "all non-major dependencies except Node.js" and "Node.js", with only a simple config like:

```json
{
  "extends": ["group:allNonMajor", "group:nodeJs"]
}
```

However, this didn't work.

Should I make my own version of the "group:allNonMajor" preset that excludes Node.js?

**Comment by rarkins** [comment, marked as answer]
To simplify the idea of grouping, you just need to understand that they need to have the same branch name.

Which Node.js updates are missing from your "all non major" PR which you want to add? e.g you want to group major Node.js updates with non-major updates of other dependencies?

**Reply by chalkygames123** [reply]
Sorry for the ambiguity. The "didn't work" in the OP meant that the Node.js update was unexpectedly included in the "all non-major" PR.

What I am expecting, more precisely, is that the PRs will be grouped as follows:

* Updates for all non-major dependencies that don't include Node.js at all (branch name: e.g. `renovate/all-minor-patch-except-node`)
* Updates for each major dependency (not grouped, each one with its own PR)
* Updates for stable Node.js (branch name: e.g. `renovate/node-15.x`)
* Updates for unstable Node.js (branch name: e.g. `renovate/node-16.x`)

I'm not sure if this can be achieved with existing presets alone at this point.

**Reply by chalkygames123** [reply]
I figured it out. I'm not sure if this is a good idea, but anyway, the above groupings have been achieved as follows.

1. Based on the Renovate's ["group:allNonMajor"](https://github.com/renovatebot/renovate/blob/4303063b2ab77e5680d96fa659dd146195f781d7/lib/config/presets/internal/group.ts#L23-L33) preset, create your own "group:allNonMajorExceptNode" preset that excludes Node.js, for example.

```json
// group:allNonMajorExceptNode.json
{
	"description": "Group all minor and patch updates except Node.js together",
	"packageRules": [
		{
			"matchPackagePatterns": ["*"],
			"matchUpdateTypes": ["minor", "patch"],
			"excludePackageNames": ["node"],
			"groupName": "all non-major dependencies except Node.js",
			"groupSlug": "all-minor-patch-except-node"
		}
	]
}
```

2. Extend it and "group:nodeJs" presets in your config (the "github>chalkygames123/renovate-config" part depends).

```json
// renovate.json
{
	"extends": [
		"github>chalkygames123/renovate-config:group:allNonMajorExceptNode",
		"group:nodeJs"
	]
}
```

3. Add a package rule to update Node.js to the latest Current release (see https://github.com/renovatebot/renovate/discussions/11180).

```json
// renovate.json
{
	"extends": [
		"github>chalkygames123/renovate-config:group:allNonMajorExceptNode",
		"group:nodeJs"
	],
	"packageRules": [
		{
			"matchPackageNames": ["node"],
			"ignoreUnstable": false
		}
	]
}
```

So here's a concrete example: https://github.com/chalkygames123/renovate-config/tree/a94e563bd0672e2d5efec5cfadc33b2c176cfc07
